### PR TITLE
Warping operator from Gardner et al., 2017

### PR DESCRIPTION
**Pull request by lefreud**
Here is an implementation of the Warping operator introducted in [Gardner et al., 2017](https://dl.acm.org/doi/10.1145/3130800.3130891). It's based on the original implementation provided by Marc-André Gardner.

Bonus: I also added a small script to make a video of a camera translation in the sphere of the environment map by using the warping operator!

https://github.com/soravux/skylibs/assets/35122940/06c8ecea-5fd5-4a2f-bbfc-cc7605578117

### CODE REVIEW

**Comment by lefreud** [comment]
Marc-André and I were also wondering if the `cachedWorldCoordinates` dictionary could instead be a cache accessed anytime the method `EnvironmentMap.worldCoordinates()` is called? By checking if the `format_` stays the same (e.g. latlong) and the panorama shape doesn't change, the method should be able to reuse a cached version of the world coordinates I think. That could save some time when we need to constantly re-compute the world coordinates, for dataset creation for instance. It might add unnecessary complexity to the code though.

**Comment by lefreud** [comment]
Thank you for the comments! I will add some tests soon. I'll also benchmark the function with/without the caching behaviour to see the performance gain.

**Comment by lefreud** [comment]
I added some unit tests and also found out that I had a sign error, which made a positive `nadir` translate the camera in the -Z direction, instead of the +Z axis. This is now fixed.

File: tools3d/warping_operator/example_warp_operator.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    originalEnvironmentMap = EnvironmentMap(args.environment, 'latlong')   
+
+    hdrScalingFactor = 0.5/np.mean(originalEnvironmentMap.data)
+    def tonemap(x):
+        return np.clip(x * hdrScalingFactor, 0, 1) ** (1/2.2)
+    
+    output_dir = args.output_dir
+    os.makedirs(output_dir, exist_ok=True)
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
```
Comment (soravux) [inline tools3d/warping_operator/example_warp_operator.py:45]: Nice use of tempfile!

**Review by soravux — APPROVED** [review]

**Comment by soravux** [comment]
Looks great, thanks a lot for this contribution!

File: tools3d/warping_operator/example_warp_operator.py (hunk 1 above)
Comment (lefreud) [reply tools3d/warping_operator/example_warp_operator.py:45]: Thanks! That was Marc-André's suggestion 😄

**Pull request merged**